zephyr/lib/alloc.c

```diff
@@ -399,15 +382,12 @@ void *rmalloc(enum mem_zone zone, uint32_t flags, uint32_t caps, size_t bytes)
 #if CONFIG_L3_HEAP
 		heap = &l3_heap;
 		/* Uncached L3_HEAP should not be used */
-		if (!zone_is_cached(zone)) {
+		if (flags & SOF_MEM_FLAG_COHERENT) {
```

Comment:
need to be careful with these. Unfortunately since we had two indicators for uncached allocations (zone and flags) their use isn't consistent everywhere. In some calls we only use a `SHARED` zone without using a flag. So those calls have to be fixed